### ISSUE TITLE
Discrepancies between code and paper

### ISSUE THREAD

**Issue by meatballs** [Axelrod-Python/axelrod-fortran#36]
Adding the authors and original rank to the characteristics dict on #37 has revealed:

- Both K74R and K74RXX have a comment saying their author was Edward Friedland. However, that name only appears once in table 2 of Axelrod's paper (ranked 61)
- K92R is TitForTat by Anatol Rapoport
- KTF2TC is not listed in table 2
- KPAVLOVC is not listed in table 2

Given that they appear in the [code for the tournament](https://github.com/Axelrod-Python/TourExec/blob/v0.2.0/src/tournament/AxTest.f#L11)  I suspect that KPAVLOVC, KTITFORTATC and KTF2TC were not participating strategies.

RANDOM is listed in table 2 and also appears in L11.

**Comment by drvinceknight** [comment]
> Given that they appear in the code for the tournament I suspect that KPAVLOVC, KTITFORTATC and KTF2TC were not participating strategies.

How neat! I guess they were perhaps written as examples for people to use to write their own.

Is there a way to see in `TourExec` if they're run by the main tournament in that?

**Comment by drvinceknight** [comment]
> Both K74R and K74RXX have a comment saying their author was Edward Friedland. However, that name only appears once in table 2 of Axelrod's paper (ranked 61)

Similarly if there's a way to see what gets run by `TourExec` perhaps we'll have some light here? (Perhaps)

**Comment by meatballs** [comment]
We need to get to grips with how the [tournament code](https://github.com/Axelrod-Python/TourExec/blob/v0.2.0/src/tournament/AxTest.f) works. Not something I've got in to yet...

**Comment by meatballs** [comment]
K74R is called at
https://github.com/Axelrod-Python/TourExec/blob/master/src/tournament/KRowFunction.f#L93

K74RXX doesn't appear at all, so I think we can conclude that K74R is the strategy by Edward Friedland in the second tournament.

**Comment by drvinceknight** [comment]
Nice detective work :+1:

**Comment by meatballs** [comment]
I think we can close this one now

**Comment by meatballs** [comment]
> K74R is called at
https://github.com/Axelrod-Python/TourExec/blob/master/src/tournament/KRowFunction.f#L93
>
> K74RXX doesn't appear at all, so I think we can conclude that K74R is the strategy by Edward Friedland in the second tournament.

Given the discovery that tourexec is not the tournament code and must, in fact, have been written after the tournament had been run, I think the conclusion above is no longer valid.

I suspect that K74RXX was the original code but that a new version was created - because of an error, perhaps, or because the author submitted an improvement. At that point, the function was renamed from K74R to K74RXX and the new code added as K74R. (This suspicion being caused by the fact that K74RXX had the variable K74R throughout, which we treated as a bug and corrected).

The interesting question is whether that change happened before of after the tournament had been run and we don't know the answer to that.

**Comment by drvinceknight** [comment]
> (This suspicion being caused by the fact that K74RXX had the variable K74R throughout, which we treated as a bug and corrected).
> 
> The interesting question is whether that change happened before of after the tournament had been run and we don't know the answer to that.

Good catch. We'll take at rerunning the tournament with both and see what the results look like.

**Comment by drvinceknight** [comment]
Looking at the source code:

- [K74R](https://github.com/Axelrod-Python/TourExec/blob/master/src/strategies/K74R.f) has the top preamble:
   ```
   FUNCTION K74R(J,M,K,L,R,JA)
   C BY EDWARD FRIEDLAND
   C TYPED BY JM 3/20/79
   c temp output
   ```

- [K74RXX](https://github.com/Axelrod-Python/TourExec/blob/master/src/strategies/K74RXX.f) has the top preamble:

    ```
    FUNCTION K74RXX(J,M,K,L,R,JA)
    C BY EDWARD FRIEDLAND
    C TYPED BY JM 3/20/79
    c k74dummy added by ax 7/22/93
    ```

This makes it look like `k74RXX` might be a dummy repeat of `K74R` , here is the diff of these two files which at a lazy first glance makes them seem like they're perhaps essentially the same:  [k74diff.txt](https://github.com/Axelrod-Python/axelrod-fortran/files/1190875/k74diff.txt)

**However these strategies are definitely different:**

```python
>>> import axelrod as axl
>>> import axelrod_fortran as axlf
>>> K74R = axlf.Player("k74r")
>>> K74RXX = axlf.Player("k74rxx")
```

Taking a look at fingerprints:

```python
>>> axl.seed(0)
>>> K74R_fp = axl.AshlockFingerprint(K74R)
>>> K74R_fp_data = K74R_fp.fingerprint(step=0.05)
>>> K74R_fp.plot()
```

![download 22](https://user-images.githubusercontent.com/2131546/28830217-0e05c134-76ce-11e7-89fe-f129aa61bf54.png)


```python
>>> axl.seed(0)
>>> K74RXX_fp = axl.AshlockFingerprint(K74RXX)
>>> K74RXX_fp_data = K74RXX_fp.fingerprint(step=0.05)
>>> K74R_fp.plot()
```

![download 23](https://user-images.githubusercontent.com/2131546/28830248-1fc08602-76ce-11e7-91ed-6adf60daf618.png)

but even simpler:

```python
>>> axl.seed(0)
>>> K74R_match = axl.Match((K74R, axl.Cooperator()), turns=3)
>>> K74R_match.play()
[(D, C), (C, C), (D, C)]
```

and:

```python
>>> axl.seed(0)
>>> K74RXX_match = axl.Match((K74RXX, axl.Cooperator()), turns=3)
>>> K74RXX_match.play()
```

My suggestion:

**TLDR**

K74R is the strategy by Edward Friedland in the second tournament.

and we don't need to modify anything.
